Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -102,33 +104,32 @@ public Task<bool> IsCompleteSubmissionAsync(string code)
             return false;
         }
 
-        public override Task<LspResponse> LspMethod(string methodName, JObject request)
+        public async Task<TextDocumentHoverResponse> Hover(HoverParams hoverParams)
         {
-            LspResponse result;
-            switch (methodName)
+            if (!hoverParams.TextDocument.Uri.TryDecodeDocumentFromDataUri(out var documentContents))
             {
-                case "textDocument/hover":
-                    // https://microsoft.github.io/language-server-protocol/specification#textDocument_hover
-                    var hoverParams = request.ToLspObject<HoverParams>();
-                    result = TextDocumentHover(hoverParams);
-                    break;
-                default:
-                    result = null;
-                    break;
+                return null;
```

Comment:
Do we want to return something other than `null`, e.g. a `TextDocumentHoverResponse.Empty`?

Reply:
A hover response needs content and an optional range and I can't think of a value that's valid according to the spec that could represent `empty` except for `null`, unless we define it like `public static TextDocumentHoverResponse Empty => null;` which doesn't buy us much.